src/nomad_parser_magres/schema_packages/workflow.py

```diff
@@ -1,27 +1,9 @@
-#
-# Copyright The NOMAD Authors.
-#
-# This file is part of NOMAD.
-# See https://nomad-lab.eu for further info.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-from nomad.metainfo import SubSection, Quantity, Reference
-from nomad.datamodel.metainfo.simulation.method import Method
+from nomad.metainfo import Quantity, Reference, SubSection
```

Comment:
This NMRMagRes class is defined to generate an extra EntryArchive if both NMR first principles and NMR magres SinglePoint EntryArchives are present. Could you please clarify how this works in linking the workflows as mentioned in the parser.py script?

Reply:
The idea's that the magres file creates an entry which connects with the CASTEP or QE one. In this entry, you will find all the details of the original DFT calculation + the output nmr properties.

The creation of such a workflow in the magres entry is just an organizational tool. This is actually what workflows should be about, just organizing in one way without cluttering entries which could be reuse for something else (in this case, the CASTEP or QE ones).

In any case, this can be changed later. We could decide to integrate the workflow better, hence just adding a few lines in the CASTEP and QE parsers to handle the possibility of a magres output file.